tests/integration/cache_test.go

```diff
@@ -485,6 +485,196 @@ func TestCacheWithPrefix(t *testing.T) {
 	}
 }
 
+func TestCacheGet(t *testing.T) {
+	integration.BeforeTest(t)
+	clus := integration.NewCluster(t, &integration.ClusterConfig{Size: 1})
+	t.Cleanup(func() { clus.Terminate(t) })
+
+	client := clus.Client(0)
+
+	c, err := cache.New(client, "", cache.WithHistoryWindowSize(32))
+	if err != nil {
+		t.Fatalf("cache.New: %v", err)
+	}
+	t.Cleanup(c.Close)
+	if err := c.WaitReady(t.Context()); err != nil {
+		t.Fatalf("cache.WaitReady: %v", err)
+	}
+
+	testGet(t, client.KV, c)
+}
+
+func TestGet(t *testing.T) {
+	integration.BeforeTest(t)
+	clus := integration.NewCluster(t, &integration.ClusterConfig{Size: 1})
+	t.Cleanup(func() { clus.Terminate(t) })
+
+	client := clus.Client(0)
+	testGet(t, client.KV, client.KV)
+}
+
+func testGet(t *testing.T, kv clientv3.KV, reader Getter) {
+	ctx := t.Context()
+	rev2PutFooA := &clientv3.Event{
+		Type: clientv3.EventTypePut,
+		Kv: &mvccpb.KeyValue{
+			Key:            []byte("/foo/a"),
+			Value:          []byte("a1"),
+			CreateRevision: 2,
+			ModRevision:    2,
+			Version:        1,
+		},
+	}
+	rev3PutFooB := &clientv3.Event{
+		Type: clientv3.EventTypePut,
+		Kv: &mvccpb.KeyValue{
+			Key:            []byte("/foo/b"),
+			Value:          []byte("b1"),
+			CreateRevision: 3,
+			ModRevision:    3,
+			Version:        1,
+		},
+	}
+	rev4PutFooC := &clientv3.Event{
+		Type: clientv3.EventTypePut,
+		Kv: &mvccpb.KeyValue{
+			Key:            []byte("/foo/c"),
+			Value:          []byte("c1"),
+			CreateRevision: 4,
+			ModRevision:    4,
+			Version:        1,
+		},
+	}
+	rev5PutFooD := &clientv3.Event{
+		Type: clientv3.EventTypePut,
+		Kv: &mvccpb.KeyValue{
+			Key:            []byte("/foo/d"),
+			Value:          []byte("d1"),
+			CreateRevision: 5,
+			ModRevision:    5,
+			Version:        1,
+		},
+	}
+	rev6DeleteFooD := &clientv3.Event{
+		Type: clientv3.EventTypeDelete,
+		Kv: &mvccpb.KeyValue{
+			Key:         []byte("/foo/d"),
+			ModRevision: 6,
+		},
+	}
+	rev7TxnPutFooA := &clientv3.Event{
+		Type: clientv3.EventTypePut,
+		Kv: &mvccpb.KeyValue{
+			Key:            []byte("/foo/a"),
+			Value:          []byte("a2"),
+			CreateRevision: 7,
+			ModRevision:    7,
+			Version:        2,
+		},
+	}
+	rev7TxnPutFooB := &clientv3.Event{
+		Type: clientv3.EventTypePut,
+		Kv: &mvccpb.KeyValue{
+			Key:            []byte("/foo/b"),
+			Value:          []byte("b2"),
+			CreateRevision: 7,
+			ModRevision:    7,
+			Version:        2,
+		},
+	}
+	rev8PutFooA := &clientv3.Event{
+		Type: clientv3.EventTypePut,
+		Kv: &mvccpb.KeyValue{
+			Key:            []byte("/foo/a"),
+			Value:          []byte("a3"),
+			CreateRevision: 8,
+			ModRevision:    8,
+			Version:        3,
+		},
+	}
+
+	tests := []struct {
+		name         string
+		key          string
+		opts         []clientv3.OpOption
+		wantKVs      map[string]string
```

Comment:
Please not only test `keyValue`, but the whole `KeyValue` including fields like `CreateRevision`, `ModRevision` and `Version`.